cfgen/cfgen

```diff
@@ -401,6 +401,19 @@ def enrich_cluster_desc(desc: Dict[str, Any], mock_p: bool) -> None:
                 float(node['facts']['memorysize_mb']))
 
         logging.debug('Node [hostname=%s] facts=%s', host, node['facts'])
+
+        portalgroup = PortalGroup
+        if 'network_ports' not in node or not bool(node['network_ports']):
+            node['network_ports'] = {port_type.name: port_type.value
+                                     for port_type in portalgroup}
+        else:
+            for key, value in node['network_ports'].items():
+                if not value:
+                    node['network_ports'][key] = portalgroup[key].value
+            for srv in [port_type.name for port_type in portalgroup]:
+                if srv not in node['network_ports']:
```

Comment:
We also need to validate that port_type.name is a valid service type, i.e. m0_server, m0_client_s3 or m0_client_other,  supported by Hare.

Reply:
`for srv in [port_type.name for port_type in portalgroup]:`
portalgroup above is as follows
```
class PortalGroup(Enum):
    hax = 22000
    m0_server = [{'name': 'confd', 'port': m0ProcT.confd},
                 {'name': 'ios', 'port': m0ProcT.ios}]
    m0_client_s3 = 22500
    m0_client_other = [{'name': 'default', 'port': 22500}]
```
So port_type.name will always be valid service type,
Here we are checking if node['network_ports'] contains entries for predefined services type in PortalGroup. If it is not present then we will use default value

Reply:
Ok